lib/deploy/activate.rb

```diff
@@ -89,6 +95,13 @@ def setup_idp_config_symlinks
       end
     end
 
+    def service_providers_file
+      return 'service_providers.sandbox' if Identity::Hostdata.domain == 'identitysandbox.gov'
+      return 'service_providers.staging' if Identity::Hostdata.env == 'staging'
+
+      'service_providers'
+    end
```

Comment:
Is this behavior that could be handled by `restrict_to_deploy_env` or something similar?

I am wary of adding hard requirements like this on filenames in another repository, those are harder to catch breakages in and if we ever rename the files or change strategies, it makes migrations complex

Reply:
Unfortunately not since we have duplicate issuers across environments and that would result in duplicate keys in the YAML file (this is why staging has its own file as much as I was originally hoping to keep staging and production together with `restrict_to_deploy_env` as the identifier. With that said, we could always update the SP configs for our internal apps so I'm open to doing that instead - it just means a bit more transition work.

As I'm thinking about it more that's probably worth it (vs depending on `Identity::Hostdata`) so I'm going to take a second stab at this. I'll ping you once it's updated!

Reply:
@zachmargolis as I'm thinking through the sandbox, any objection to all sandbox integrations having the same issuer regardless of the environment (e.g. `dev` and `int` would be the same)? I don't think it's a big deal but just double-checking.

Reply:
I think that's fine, I believe we've reused issuer names across sandbox envs in the past as well